utils/rpmlint

```diff
@@ -1,3 +1,2 @@
 addFilter("setup-not-quiet")
 addFilter("invalid-url Source0")
-addFilter("macro-in-comment")
```

Comment:
I can see two positions, here.  The first is that macros in comments don't hurt anything and therefore we don't care and do not wish to be warned about them...in which case, this change is good.  The second is that we no longer have any macros in comments, and therefore this warning doesn't make any difference...but, if we want to avoid macros in comments in the future, then we should not be removing this, and this change would be bad.

My understanding is that macros are expanded even if they are in comments; that their expansions can have multiple lines which can escape from the comment where they are invoked and cause problems; and that this is a good and valuable thing to check for in the general case (even though its annoying because it seems handy to be able to refer to the macros in a comment).

So...do we really want to delete this line?

Reply:
This is a back-port of PR #2882 as it is on `main`.  If we want to re-add the warning, we can add it back, but it should be addressed on `main` first.

Reply:
> If we want to re-add the warning, we can add it back, but it should be addressed on main first.

"First"?...as in before this PR is merged??

Nevertheless, if this is a "bad" change, then we should re-add it on main and leave this branch alone.  (Why would we remove it from here and then re-add it here after updating `main`?  Surely we don't have any requirement to keep this branch in that kind of lock-step with `main`....)

Reply:
Because I am trying to get this to build without warnings as we have on `main`.

Reply:
FWIW, this is what happens when this is present:

```
$ make srpm
make[1]: Entering directory '/home/pportant/Documents/pbench/agent/rpm'
Building Makefile for pbench-agent-0.72.0 from /home/pportant/rpmbuild/TMP/pbench-agent-0.72.0 to /home/pportant/rpmbuild
make[1]: Nothing to be done for 'Makefile'.
make[1]: Leaving directory '/home/pportant/Documents/pbench/agent/rpm'
make[1]: Entering directory '/home/pportant/Documents/pbench/agent/rpm'
Building srpm for pbench-agent-0.72.0 from /home/pportant/rpmbuild/TMP/pbench-agent-0.72.0 to /home/pportant/rpmbuild
mkdir -p /home/pportant/rpmbuild/BUILD /home/pportant/rpmbuild/BUILDROOT /home/pportant/rpmbuild/SPECS /home/pportant/rpmbuild/SOURCES /home/pportant/rpmbuild/SRPMS /home/pportant/rpmbuild/RPMS /home/pportant/rpmbuild/TMP
if [ -e ./seqno ] ;then expr 1 + 1 > ./seqno ;fi
jinja2 pbench-agent.spec.j2 -D version=0.72.0 -D gdist=g6346e1350 -D seqno=1 > /home/pportant/rpmbuild/SPECS/pbench-agent.spec
cp /home/pportant/Documents/pbench/utils/rpmlint /home/pportant/rpmbuild/SPECS/pbench-common.rpmlintrc
XDG_CONFIG_HOME=/home/pportant/Documents/pbench/utils rpmlint /home/pportant/rpmbuild/SPECS/pbench-agent.spec
================================================================================= rpmlint session starts ================================================================================
rpmlint: 2.2.0
configuration:
    /usr/lib/python3.10/site-packages/rpmlint/configdefaults.toml
    /etc/xdg/rpmlint/fedora.toml
    /etc/xdg/rpmlint/licenses.toml
    /etc/xdg/rpmlint/scoring.toml
    /etc/xdg/rpmlint/users-groups.toml
    /etc/xdg/rpmlint/warn-on-functions.toml
rpmlintrc: /home/pportant/rpmbuild/SPECS/pbench-common.rpmlintrc
checks: 32, packages: 1

pbench-agent.spec: E: unused-rpmlintrc-filter "macro-in-comment"
================================================== 0 packages and 1 specfiles checked; 1 errors, 0 warnings, 1 badness; has taken 0.0 s =================================================
make[1]: *** [../../utils/rpm.mk:85: spec] Error 64
make[1]: Leaving directory '/home/pportant/Documents/pbench/agent/rpm'
make: *** [Makefile:51: srpm] Error 2
```

Reply:
OK, after flailing around on the internet, I can't figure out what that error is really supposed to mean.  (I think it means that the `"macro-in-comment"` filter doesn't apply to the scenario that we're running in...but that doesn't make sense, either.)  Since I cannot convince myself that that filter is supposed to do what we want it to, and since having it is doing something that we _don't_ want it to, I guess removing it is a good thing.  🍤